### ISSUE TITLE
MD_identifier in metadata document and service metadata

### ISSUE THREAD

**Issue by Dnoack49** [INSPIRE-MIF/helpdesk-validator#638]
Dear community,

we have tested the following download service (direct WFS):
https://inspire.brandenburg.de/services/am_btf_wfs?ACCEPTVERSIONS=2.0.0&request=GetCapabilities&service=WFS&VERSION=2.0.0

for which we receive this error:
![grafik](https://user-images.githubusercontent.com/8046960/138232627-77d4232c-3f68-4c47-909d-d2dfab1bd78e.png)

The relevant metadata UUIDs are:

- dataset: _9780015C-CB95-4DA6-BF34-326A5E523EEE_ 
- datasource: _328199e4-1fe7-4193-b05e-91d699238d3b_
- WFS: _6A6D92AC-0E41-45AE-89AF-33EE86CA5232_

The service metadata can be found here:
https://www.metaver.de/csw?REQUEST=GetRecordById&SERVICE=CSW&VERSION=2.0.2&id=6A6D92AC-0E41-45AE-89AF-33EE86CA5232

Contrary to the error message the MD_Identifier https://registry.gdi-de.org/id/de.bb.metadata/328199e4-1fe7-4193-b05e-91d699238d3b is listed in the service metadata

- under the GetCapabilities-/GetFeature-Operation:
![grafik](https://user-images.githubusercontent.com/8046960/138235066-e18e9adc-bf52-4ee4-9397-52d1b6dc2877.png)

- under the operatesOn-Element for the coupling:
![grafik](https://user-images.githubusercontent.com/8046960/138235331-a3d2a04f-b39b-4db8-8f57-a7f9c4fbc7ec.png)

In the service metadata the identifier _9780015C-CB95-4DA6-BF34-326A5E523EEE_ is indeed listed and refers to the essential data-service coupling (see above mentioned operatesOn-element).
As we cannot locate the problem neither with the metadata document nor the service metadata, we would appreciate your feedback.

**Comment by dperezBM** [comment]
Dear @Dnoack49,

Thank you for your message. We will analyze the issue that you reported to us and we will come back to you as soon as we get any further information.

Best regards.

**Comment by fabiovin** [comment]
Dear @Dnoack49,

I had a look at your resources and at the dataset-service linkage implementation.

I would suggest trying to remove the _uuidref_  optional attribute from the _operatesOn_ element.

in addition, since you used the _gmd:MD_Identifier_ element for the "Unique resource identifier" (it include only the code element), 

```xml
<gmd:identifier>
    <gmd:MD_Identifier>
       <gmd:code>
               <gco:CharacterString>https://registry.gdi-de.org/id/de.bb.metadata/328199e4-1fe7-4193-b05e- 
                        91d699238d3b</gco:CharacterString>
       </gmd:code>
     </gmd:MD_Identifier>
</gmd:identifier>
```

you should use this value for the code element in the WFS Extended capabilities, even if both Code and Namespace elements are foreseen

```xml
<inspire_dls:SpatialDataSetIdentifier>
     <inspire_common:Code>https://registry.gdi-de.org/id/de.bb.metadata/328199e4-1fe7-4193-b05e- 
                        91d699238d3b</inspire_common:Code>
     <inspire_common:Namespace>https://registry.gdi-de.org/id/de.bb.metadata</inspire_common:Namespace>
</inspire_dls:SpatialDataSetIdentifier>
```

**Comment by Dnoack49** [comment]
Dear @fabiovin , 

thank you for your feedback. For now I have altered the Extended Capabilities.
Here the [GetCapabilities](https://inspire.brandenburg.de/services/am_btf_wfs?service=WFS&request=GetCapabilities)

![grafik](https://user-images.githubusercontent.com/8046960/138900901-30570df8-b478-4f42-95b5-0aa52d7776b9.png)

Today I could only run the WFS-tests on the staging instance. Yet after having implemented your suggestion I still get the same error message as above.

Unfortunately for this webservice I cannot remove the _uuidref_  from the service metadata as I do not have access to the customers metadata system. I normally have to contact the person in charge and such changes usually take quite some time (in fact too much for just a quick test like this). However, what I could do is to take another WFS for which I have the authority alter the metadata. I will get back to you with that soon.

Kind regards

**Comment by Dnoack49** [comment]
Dear @fabiovin 
I have been trying to test the a wfs but the validator does not start. I receive an error on both - the production and staging instance ("an error has occured"). Do you have an idea when the validator will be up again?

**Comment by fabiovin** [comment]
Dear @Dnoack49,

both instances are working again.

Let me know about the test to remove the _uuidref_ optional attribute from the _operatesOn_ element.

In the meanwhile, we are investigating the ETS if it is the problem.

**Comment by Dnoack49** [comment]
Dear @fabiovin ,

so I have carried out the test with another WFS

Capabilities: https://inspire.brandenburg.de/services/am_ugk300_wfs?service=WFS&request=GetCapabilities
Service metadata: https://geoportal.brandenburg.de/gs-json/xml?fileid=68117975-02c0-40a6-b6ca-cb9a0dce89e7

In this case the optional _uuidref_ attribute is not listed in the operatesOn element:
![image](https://user-images.githubusercontent.com/8046960/140037219-f19c23ef-c6f1-4fc0-b8f9-c617809e2776.png)

And the WFS Extended Capabilities were altered as you suggested:
![image](https://user-images.githubusercontent.com/8046960/140037396-3c85954d-7823-48ef-b416-b22b154d5279.png)

While in the production instance, all test cases in the Capabilities module were skipped (due to problems with other test cases) I receive the following error in the staging instance:
![image](https://user-images.githubusercontent.com/8046960/140037524-de4039d6-c6bc-4179-9197-b27360d6ea54.png)

I assume that the error in this case is the reference of the dataset id (68117975-02c0-40a6-b6ca-cb9a0dce89e7) in the MD_identifier elements?

**Comment by Dnoack49** [comment]
Dear @fabiovin ,
Have you had the chance to take a look at the error message above?
kind regards

**Comment by fabiovin** [comment]
Dear @Dnoack49,

we analyzed the problem and found that the validator is looking for the operatesOn/@uuidref attribute, but since it is optional it should refer to the operatesOn /xlink:href attribute.

We are going to implement the change in the staging instance and will let you know when the fix is ready for testing.

**Comment by Dnoack49** [comment]
Dear @fabiovin , 

I have tested the service again and the error seems to be fixed on the staging but not yet in the production instance, correct? Will this fix be part of the [release v2022.0](https://github.com/INSPIRE-MIF/helpdesk-validator/tree/master/release%20strategy#v20220-15-01-2022)?

Kind regards

**Comment by arantzaetxebarria** [comment]
Dear @Dnoack49 

Thanks a lot for your confirmation. Yes, this fix will be part of the release v2022.0.

Regards,

Arantza
